### PR TITLE
refactor: use jiff crate replace chrono in date/timestamp func

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

In jiff https://docs.rs/jiff/latest/jiff/index.html#time-zone-features, we enable `tzdb-bundle-always`(When enabled, Jiff will forcefully depend on the jiff-tzdb crate). It's not a good idea when use jiff crate.

But if not enable it, jiff will read tzinfo from file /usr/share/zoneinfo. This file relate `tzdata` in system env. And we init timezone in  function ctx, if system not install tzdata will cause panic. So we enable feature `tzdb-bundle-always` as default.

Done:

register_diff_functions(registry)
register_to_number_functions(registry)
register_add_functions(registry)
register_sub_functions(registry)
register_timestamp_to_date
register_date_to_timestamp
register_to_string
register_string_to_date
register_string_to_timestamp
register_real_time_functions
register_rounder_functions

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16787)

### CODE REVIEW

File: src/query/functions/src/scalars/datetime.rs [hunk 1]
```diff
@@ -614,24 +621,22 @@ fn register_string_to_date(registry: &mut FunctionRegistry) {
...
-        |ctx, domain| {
-            let tz = ctx.tz.tz;
+        |ctx, _| {
             FunctionDomain::Domain(NullableDomain {
                 has_null: false,
                 value: Some(Box::new(SimpleDomain {
-                    min: calc_timestamp_to_date(domain.min, tz),
-                    max: calc_timestamp_to_date(domain.max, tz),
+                    min: calc_timestamp_to_date(-30610224000000000, ctx.jiff_tz.clone()),
+                    max: calc_timestamp_to_date(253402207200000000, ctx.jiff_tz.clone()),
```
Comment (TCeason) [inline src/query/functions/src/scalars/datetime.rs:639]: max second value in jiff

File: src/query/expression/src/function.rs [hunk 2]
```diff
@@ -98,7 +98,8 @@ pub enum FunctionEval {
 #[derive(Clone)]
 pub struct FunctionContext {
     pub tz: TzLUT,
```
Comment (andylokandy) [inline src/query/expression/src/function.rs:100]: Can we remove it?

Comment (TCeason) [reply src/query/expression/src/function.rs:100]: Cannot. Date_format need this. Jeff not support format string with no ascii format

File: src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs [hunk 3]
```diff
@@ -103,26 +219,22 @@ where T: AsRef<[u8]>
...
-            })?;
 
-        let less_1000 = |dt: DateTime<Tz>| {
+        let d = v.parse::<Date>().map_err_to_code(ErrorCode::BadBytes, || {
+            format!(
+                "Date Parsing Error: The value '{}' could not be parsed into a valid Date",
+                v
+            )
+        })?;
+        let less_1000 = |dt: Zoned| {
```
[the remark this answers is not in the document]
Comment (TCeason) [reply src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs:229]: Yes jiff date years support `-999 ~ 999`. But year less than 1000 maybe no any sense. So I reserve this logic.

File: src/query/expression/src/function.rs (hunk 2 above)
Comment (TCeason) [reply src/query/expression/src/function.rs]: > Can we remove it? Or convert from jiff_tz?

remove.

**Review by b41sh — APPROVED** [review]

**Comment by TCeason** [comment]
Please review this pr @sundy-li

**Review by sundy-li — APPROVED** [review]

**Pull request merged**